app/src/main/kotlin/com/intive/picover/parties/view/PartiesScreen.kt

```diff
@@ -29,30 +30,38 @@ import androidx.navigation.NavHostController
 import com.intive.picover.common.error.PicoverGenericError
 import com.intive.picover.common.loader.PicoverLoader
 import com.intive.picover.main.theme.Typography
+import com.intive.picover.parties.model.PartiesEvent
+import com.intive.picover.parties.model.PartiesSideEffect.NavigateToPartyDetails
 import com.intive.picover.parties.model.Party
 import com.intive.picover.parties.viewmodel.PartiesViewModel
+import kotlinx.coroutines.flow.collectLatest
 
 @Composable
 fun PartiesScreen(
 	viewModel: PartiesViewModel,
 	navController: NavHostController,
 ) {
 	val state by viewModel.state
+	LaunchedEffect(true) {
+		viewModel.sideEffect.collectLatest { effect ->
+			when (effect) {
+				is NavigateToPartyDetails -> navController.navigate("partyDetails/${effect.partyId}")
```

Comment:
I'm not convinced that this is the right place. Shouldn't we have some kind of reducer? Where are tests for that?

Reply:
Yeah i know, for now i didn't discover nothing better...

Reply:
Imo that's the good starting point. Later on, we could add another layer responsible for navigation. The easiest option for now would be to just store `navController` in some navigator and navigate from there, maybe this way it would be easier to test it as it will be more separated from the view. As far as I know, if we are using a single activity and Compose all the Compose screens have the same `navController` so we only need some kind of "NavControllerHolder" which will store a proper instance of `navController` (attach and detach in activity).